.github/workflows/ci.yml

```diff
@@ -201,64 +205,65 @@ jobs:
-#        shell: powershell
-#        run: |
-#          cri-containerd.test.exe -cri-endpoint="npipe://./pipe/containerd-containerd" -feature="WCOWProcess" -feature="HostProcess"
+  #      Enable these tests once the required JobContainer images are updated.
+  #
+  #      - name: Install containerd service
+  #        shell: powershell
+  #        run: |
+  #          mkdir C:\containerd
+  #          Set-Content C:/containerd/containerd.toml @"
+  #          version = 2
+  #          [plugins]
+  #              [plugins."io.containerd.grpc.v1.cri".containerd]
+  #                default_runtime_name = "runhcs-wcow-process"
+  #                disable_snapshot_annotations = false
+  #                discard_unpacked_layers = false
+  #                ignore_blockio_not_enabled_errors = false
+  #                ignore_rdt_not_enabled_errors = false
+  #                no_pivot = false
+  #                snapshotter = "windows"
+  #
+  #                [plugins."io.containerd.grpc.v1.cri".containerd.runtimes]
+  #
+  #                [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runhcs-wcow-hypervisor]
+  #                  runtime_type = "io.containerd.runhcs.v1"
+  #                  [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runhcs-wcow-hypervisor.options]
+  #                    Debug = true
+  #                    DebugType = 2
+  #                    SandboxPlatform = "windows/amd64"
+  #                    SandboxIsolation = 1
+  #
+  #                  [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runhcs-wcow-process]
+  #                    runtime_type = "io.containerd.runhcs.v1"
+  #                    pod_annotations = ["microsoft.com/*", "io.microsoft.*" ]
+  #                    container_annotations = ["microsoft.com/*", "io.microsoft.*" ]
+  #
+  #                    [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runhcs-wcow-process.options]
+  #          "@
+  #
+  #          containerd.exe --register-service --log-level=debug --config C:/containerd/containerd.toml --service-name containerd --address //./pipe/containerd-containerd --state C:/ProgramData/containerd/state --root C:/ProgramData/containerd/root --log-file C:/containerd/containerd.log
+  #          Set-Service containerd -StartupType Automatic
+  #          Start-Service containerd
+  #
+  #      - name: Build test binary
+  #        working-directory: src/github.com/Microsoft/hcsshim/test
+  #        shell: powershell
+  #        run: |
+  #          go test -mod=mod -o "${{ github.workspace }}/bin/cri-containerd.test.exe" -gcflags=all=-d=checkptr -c ./cri-containerd/ -tags functional
+  #
+  #      - name: Run hcsshim integration tests
+  #        shell: powershell
+  #        run: |
+  #          cri-containerd.test.exe -cri-endpoint="npipe://./pipe/containerd-containerd" -feature="WCOWProcess" -feature="HostProcess"
 
   build:
-    runs-on: 'windows-2019'
+    needs: [test, integration-tests]
```

Comment:
Since the only thing this job does is publish artifacts, i figured they only be uploaded if tests pass
I am not sure what else this stage would get us, since if the binaries dont compile, the test jobs would fail as well